Fix mypy issue with pytorch nightly

## 🐛 Bug description

Currently, there is an issue with running mypy on ignite when using pytorch-nightly : 
```
 ignite/distributed/comp_models/native.py:334: error: Incompatible types in
assignment (expression has type "ReduceOp", variable has type "str") 
[assignment]
                op = self._reduce_op_map[op]
                     ^
Found 1 error in 1 file (checked 84 source files)
```

https://github.com/pytorch/ignite/runs/1373059337#step:5:17

@gruebel could you please take a look if we can fix it ? Thanks

--- a/ignite/distributed/comp_models/native.py
+++ b/ignite/distributed/comp_models/native.py
@@ -331,8 +331,8 @@ if has_native_dist_support:
         def _do_all_reduce(self, tensor: torch.Tensor, op: str = "SUM") -> torch.Tensor:
             if op not in self._reduce_op_map:
                 raise ValueError("Unsupported reduction operation: '{}'".format(op))
-            op = self._reduce_op_map[op]
-            dist.all_reduce(tensor, op)
+            reduce_op = self._reduce_op_map[op]
+            dist.all_reduce(tensor, reduce_op)
             return tensor
 
         def _do_all_gather(self, tensor: torch.Tensor) -> torch.Tensor:

--- a/ignite/handlers/checkpoint.py
+++ b/ignite/handlers/checkpoint.py
@@ -4,7 +4,7 @@ import os
 import tempfile
 import warnings
 from abc import ABCMeta, abstractmethod
-from collections import OrderedDict, namedtuple
+from collections import OrderedDict
 from tempfile import _TemporaryFileWrapper  # type: ignore[attr-defined]
 from typing import Any, Callable, Dict, List, Mapping, NamedTuple, Optional, Tuple, Union
 

